### PR TITLE
Request Service

### PR DESCRIPTION
**Pull request by BRHoward**
Starting front end refactor by putting all HTTP requests into a separate service.

### CODE REVIEW

**Comment by BRHoward** [comment]
@jebbinSCL @stevenhillcox @sievins

File: public/js/ChatController.js [hunk 1]
```diff
@@ -25,25 +25,12 @@
...
-                .ok("Done")
-                .cancel("Cancel");
-            $mdDialog.show(confirm).then(function (newName) {
-                updateConversationDetails(conversationid, newName);
-            });
-        }
-
         function loadUserInfo() {
-            $http.get("/api/user")
+            RequestService.get_userInfo()
```
Comment (jebbinSCL) [inline public/js/ChatController.js]: The underscores are necessary, just getUserInfo is good

Comment (jebbinSCL) [inline public/js/ChatController.js]: Same for everywhere else below.

**Comment by jebbinSCL** [comment]
Only change is to remove the underscores from the function names.

**Comment by jebbinSCL** [comment]
:ship:

**Pull request merged**